content/articles/2017-12-tibble-1-4-1.Rmd

```diff
@@ -0,0 +1,93 @@
+---
+title: tibble 1.4.1
+slug: tibble-1-4-1
+description: >
+    Tibbles are a modern reimagining of the data frame, keeping what time has shown to be effective, and throwing out what is not, with nicer default output too! This article showcases the changes in the most recent version.
+date: 2017-12-29
+author: Kirill Müller, Hadley Wickham
+photo:
+  url: https://unsplash.com/photos/cFeRWofFkw8
+  author: Patrick Schneider
+categories: [package]
+---
+
+```{r include = FALSE}
+library(tibble)
+```
+
+We’re excited to announce version 1.4.1 of the tibble package. Tibbles are a modern reimagining of the data frame, keeping what time has shown to be effective, and throwing out what is not, with nicer default output too! Grab the latest version with:
+
+```r
+install.packages("tibble")
+```
+
+This release contains the following major changes:
+
+- Colored terminal output
+- Compatibility fixes for code that expects data frames
+- Improved `add_column()`
+
+There are many other small improvements and bug fixes: please see the [release notes](https://github.com/tidyverse/tibble/releases/tag/v1.4.1) for a complete list.
+
+Thanks to [Anh Le](https://github.com/LaDilettante) for the `add_cases()` alias, to [Davis Vaughan](https://github.com/DavisVaughan) for improving `add_column()`, to [Patrick O. Perry](https://github.com/patperry) for converting C++ code to C, and to all the other contributors. Use the [issue tracker](https://github.com/tidyverse/tibble/issues) to submit bugs or suggest ideas, your contributions are always welcome.
+
+## Colored terminal output
+
+The most important change of this release is the use of color and other markup for the output in the terminal:
+
+![](/images/tibble-1-4-1/screenshot.png)
+
+Also, the output routine has been optimized to display more data if the horizontal space is insufficient to show everything.
+
+For numbers, the three significant digits are highlighted, negative numbers are shown in red. Missing values have a green background, and character values and factors are abbreviated if necessary. The highlighting is implemented by the new pillar package, which also offers extension points for packages such as `hms` that provide custom data types. See the [Extending tibble](https://cran.r-project.org/web/packages/tibble/vignettes/extending.html) vignette for details.
```

Comment:
Can you add something like:

> We're still figuring out how to use colour most effectively, so expect this to change in minor ways in the in future. If you have feedback, please let us know by filing an issue in the pillar package.

Reply:
And I think make a bulleted list?

* Numeric columns have two displays. If there is room, the full number is displayed, making it easy to make order of magnitude comparisons. Digits after the first three are dimmed to emphasise the important components. Negative numbers are shown in red.

* Missing values get a yellow background. This makes it easier to distinguish the string "NA" from a missing value.

* Long character values and factors are abbreviated in order to fit more columns on screen.